addons/account/models/account_journal.py

```diff
@@ -323,6 +331,40 @@ def _compute_suspense_account_id(self):
             else:
                 journal.suspense_account_id = False
 
+    def _inverse_type(self):
+        # Create an alias for purchase/sales journals
+        for journal in self:
+            if journal.type not in ('purchase', 'sale'):
+                if journal.alias_id:
+                    journal.alias_id.sudo().unlink()
+                continue
+
+            alias_name = next(string for string in (
+                journal.alias_name,
+                journal.name,
+                journal.code,
+                journal.type,
+            ) if string and is_encodable_as_ascii(string))
+
+            if journal.company_id != self.env.ref('base.main_company'):
+                alias_name = f"{alias_name}-{journal.company_id.name}"
```

Comment:
It wasn't perfect before either, but the company name needs to be ascii encodable too